src/colmap/estimators/cost_functions.h

```diff
@@ -349,182 +349,192 @@ inline void EigenQuaternionToAngleAxis(const T* eigen_quaternion,
-            new AbsolutePoseErrorCostFunctor(cam_from_world, covariance_cam)));
+        new ceres::AutoDiffCostFunction<AbsolutePosePriorCostFunctor, 6, 4, 3>(
+            new AbsolutePosePriorCostFunctor(cam_from_world_prior,
+                                             cam_from_world_prior_cov)));
   }
 
   template <typename T>
-  bool operator()(const T* const cam_from_world_q,
-                  const T* const cam_from_world_t,
+  bool operator()(const T* const cam_from_world_rotation,
+                  const T* const cam_from_world_translation,
                   T* residuals_ptr) const {
-    const Eigen::Quaternion<T> param_from_measured_q =
-        EigenQuaternionMap<T>(cam_from_world_q) *
-        world_from_cam_.rotation.cast<T>();
-    EigenQuaternionToAngleAxis(param_from_measured_q.coeffs().data(),
+    const Eigen::Quaternion<T> param_from_prior_rotation =
+        EigenQuaternionMap<T>(cam_from_world_rotation) *
+        world_from_cam_prior_.rotation.cast<T>();
+    EigenQuaternionToAngleAxis(param_from_prior_rotation.coeffs().data(),
                                residuals_ptr);
 
-    Eigen::Map<Eigen::Matrix<T, 3, 1>> param_from_measured_t(residuals_ptr + 3);
-    param_from_measured_t = EigenVector3Map<T>(cam_from_world_t) +
-                            EigenQuaternionMap<T>(cam_from_world_q) *
-                                world_from_cam_.translation.cast<T>();
+    Eigen::Map<Eigen::Matrix<T, 3, 1>> param_from_prior_translation(
+        residuals_ptr + 3);
+    param_from_prior_translation =
+        EigenVector3Map<T>(cam_from_world_translation) +
+        EigenQuaternionMap<T>(cam_from_world_rotation) *
+            world_from_cam_prior_.translation.cast<T>();
 
     Eigen::Map<Eigen::Matrix<T, 6, 1>> residuals(residuals_ptr);
-    residuals.applyOnTheLeft(sqrt_information_cam_.template cast<T>());
+    residuals.applyOnTheLeft(
+        cam_from_world_prior_sqrt_info_.template cast<T>());
     return true;
   }
 
  private:
-  const Rigid3d world_from_cam_;
-  const EigenMatrix6d sqrt_information_cam_;
+  const Rigid3d world_from_cam_prior_;
+  const Eigen::Matrix6d cam_from_world_prior_sqrt_info_;
 };
 
-// 6-DoF error between two absolute poses based on a measurement that is their
-// relative pose, with identical scale for the translation. The covariance is
-// defined in the reference frame of the camera j.
-// Its first and last three components correspond to the rotation and
-// translation errors, respectively.
-//
-// Derivation:
-// j_T_w = ΔT_j·j_T_i·i_T_w
-// where ΔT_j = exp(η_j) is the residual in SE(3) and η_j in tangent space.
-// Thus η_j = log(j_T_w·i_T_w⁻¹·i_T_j)
-// Rotation term: ΔR = log(j_R_w·i_R_w⁻¹·i_R_j)
-// Translation term: Δt = j_t_w + j_R_w·i_R_w⁻¹·(i_t_j -i_t_w)
-struct MetricRelativePoseErrorCostFunctor {
+// 3-DoF error on the camera position in the world coordinate frame.
+struct AbsolutePosePositionPriorCostFunctor {
```

Comment:
The mixed-up diffs make it very difficult to check for changes, would you mind splitting this into 2 PRs for reordering the classes vs renaming all variables?